server/src/main/java/org/opensearch/search/backpressure/settings/SearchShardTaskSettings.java

```diff
@@ -52,7 +193,114 @@ public long getTotalHeapBytesThreshold() {
+    }
+
+    public double getHeapVarianceThreshold() {
+        return heapVarianceThreshold;
+    }
+
+    public int getHeapMovingAverageWindowSize() {
+        return heapMovingAverageWindowSize;
+    }
+
+    public void setTotalHeapPercentThreshold(double totalHeapPercentThreshold) {
         this.totalHeapPercentThreshold = totalHeapPercentThreshold;
     }
+
+    public void setCpuTimeMillisThreshold(long cpuTimeMillisThreshold) {
+        this.cpuTimeMillisThreshold = cpuTimeMillisThreshold;
+    }
+
+    public void setElapsedTimeMillisThreshold(long elapsedTimeMillisThreshold) {
+        this.elapsedTimeMillisThreshold = elapsedTimeMillisThreshold;
+    }
+
+    public void setHeapPercentThreshold(double heapPercentThreshold) {
+        this.heapPercentThreshold = heapPercentThreshold;
+    }
+
+    public void setHeapVarianceThreshold(double heapVarianceThreshold) {
+        this.heapVarianceThreshold = heapVarianceThreshold;
+    }
+
+    public void setHeapMovingAverageWindowSize(int heapMovingAverageWindowSize) {
+        this.heapMovingAverageWindowSize = heapMovingAverageWindowSize;
+    }
+
+    public double getCancellationRatio() {
+        return cancellationRatio;
+    }
+
+    private void setCancellationRatio(double cancellationRatio) {
+        this.cancellationRatio = cancellationRatio;
+        notifyListeners(listener -> listener.onRatioChanged(cancellationRatio));
+    }
+
+    public double getCancellationRate() {
+        return cancellationRate;
+    }
+
+    public double getCancellationRateNanos() {
+        return getCancellationRate() / TimeUnit.MILLISECONDS.toNanos(1); // rate per nanoseconds
+    }
+
+    private void setCancellationRate(double cancellationRate) {
+        this.cancellationRate = cancellationRate;
+        notifyListeners(listener -> listener.onRateChanged(cancellationRate));
+    }
+
+    public double getCancellationBurst() {
+        return cancellationBurst;
+    }
+
+    private void setCancellationBurst(double cancellationBurst) {
+        this.cancellationBurst = cancellationBurst;
+        notifyListeners(listener -> listener.onBurstChanged(cancellationBurst));
+    }
+
+    public List<TaskResourceUsageTracker> getTrackers() {
+        List<TaskResourceUsageTracker> trackers = new ArrayList<>();
+        trackers.add(new CpuUsageTracker(this::getCpuTimeNanosThreshold));
+        if (HEAP_SIZE_BYTES > 0) {
+            trackers.add(
+                new HeapUsageTracker(
+                    this::getHeapVarianceThreshold,
+                    this::getHeapBytesThreshold,
+                    this.getHeapMovingAverageWindowSize(),
+                    clusterSettings,
+                    SETTING_HEAP_MOVING_AVERAGE_WINDOW_SIZE
+                )
+            );
+        }
+        trackers.add(new ElapsedTimeTracker(this::getElapsedTimeNanosThreshold, System::nanoTime));
+        return Collections.unmodifiableList(trackers);
+    }
```

Comment:
Why does it belong here? It isn't a setting.

Reply:
Moved it to 'SearchBackpressureService`.